src/main/java/Exceptions/APIException.java

```diff
@@ -0,0 +1,9 @@
+package Exceptions;
+
+import org.springframework.http.HttpStatus;
+
+public abstract interface APIException {
```

Comment:
`abstract` is implicit for java interfaces and thus redundant here.
Please remove it.